pkg/domain/stateDiffConfig.go

```diff
@@ -0,0 +1,61 @@
+package domain
+
+import "github.com/cloudogu/k8s-blueprint-operator/pkg/domain/common"
+
+type CombinedDoguConfigDiff struct {
+	DoguConfigDiff          DoguConfigDiff
+	SensitiveDoguConfigDiff SensitiveDoguConfigDiff
+}
+
+type DoguConfigDiff []DoguConfigEntryDiff
+type SensitiveDoguConfigDiff []SensitiveDoguConfigEntryDiff
+type GlobalConfigDiff []GlobalConfigEntryDiff
+
+type DoguConfigValueState ConfigValueState
+type EncryptedDoguConfigValueState ConfigValueState
+type GlobalConfigValueState ConfigValueState
+
+type DoguConfigEntryDiff struct {
+	Key      common.DoguConfigKey
+	Actual   DoguConfigValueState
+	Expected DoguConfigValueState
+	Action   ConfigAction
+}
+
+type SensitiveDoguConfigEntryDiff struct {
+	Key      common.SensitiveDoguConfigKey
+	Actual   EncryptedDoguConfigValueState
+	Expected EncryptedDoguConfigValueState
+	Action   ConfigAction
+}
+
+type GlobalConfigEntryDiff struct {
+	Key      common.GlobalConfigKey
+	Actual   GlobalConfigValueState
+	Expected GlobalConfigValueState
+	Action   ConfigAction
+}
+
+type ConfigValueState struct {
+	Value  string
+	Exists bool
+}
+
+type ConfigAction string
+
+const (
+	ConfigActionNone   ConfigAction = "none"
+	ConfigActionSet    ConfigAction = "set"
+	ConfigActionRemove ConfigAction = "remove"
+)
+
+func getNeededConfigAction(expected ConfigValueState, actual ConfigValueState) ConfigAction {
```

Comment:
🚫 **[checkstyle]** <unused> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>func `getNeededConfigAction` is unused